internal/server/Server.go

```diff
@@ -14,6 +14,9 @@ type Server struct {
 	baseserver.BaseServer[objects.IBaseBiker]
 	lootBoxes map[uuid.UUID]*objects.LootBox
 	megaBikes map[uuid.UUID]*objects.MegaBike
+	// megaBikeRiders is a mapping from Agent ID -> ID of the bike that they are riding
+	// helps with efficiently managing ridership status
+	megaBikeRiders map[uuid.UUID]uuid.UUID
 }
 
 func Initialize(iterations int) baseserver.IServer[objects.IBaseBiker] {
```

Comment:
I'd declare a separate server interface, `IMegaBikeServer` say, that you can operate on. It allows you to compose the `IServer` interface, improving clarity since you don't have to consistently call it as you currently are (`baseserver.IServer[IBaskeBiker]`)

You can then implement this server interface with some Server class

Reply:
(i.e., with your `Initialise` method)